org-code-javabuilder/theater/src/main/java/org/code/theater/Theater.java

```diff
@@ -0,0 +1,202 @@
+package org.code.theater;
+
+public enum Instrument {
+   PIANO, BASS
+}
+
+public static class Theater {
+   /**
+    * Returns the width of the theater canvas. Right now this will always be
+    * 400 pixels.
+    */
+   public static double getWidth() {
+   }
+   
+   /**
+    * Returns the height of the theater canvas. Right now this will always be
+    * 400 pixels.
+    */
+   public static double getHeight() {
+   }
+   
+   /**
+    * Plays the array of samples provided.
+    * 
+    * @param sound an array of samples to play.
+    */
+   public static void playSound(double[] sound) {
+   }
+
+   /**
+    * Plays the sound referenced by the file name.
+    * 
+    * @param filename the file to play in the asset manager.
+    * @throws FileNotFoundException if the file can't be found in the project.
+    */
+   public static void playSound(String filename) throws FileNotFoundException {
+   }
+
+   /**
+    * Plays a note with the selected instrument.
+    * 
+    * @param instrument the instrument to play.
+    * @param note       the note to play. 60 represents middle C on a piano.
+    * @param seconds    length of the note. Implementer's note: Behind the scenes,
+    *                   this should just be implemented using an array of loopable
+    *                   sounds (Mike can generate these).
+    */
+   public static void playNote(Instrument instrument, int note, double seconds) {
+   }
+
+   /**
+    * Wait the provided number of seconds before performing the next draw or play
+    * command.
+    * 
+    * @param seconds The number of seconds to wait. This can be a fraction of a
+    *                second, but the smallest value can be .1 seconds.
+    */
+   public static void wait(double seconds) {
+   }
+
+   /**
+    * Clear everything, starting from the beginning and emptying the canvas.
+    */
+   public static void reset() {
+   }
+
+   /**
+    * Draw an image on the canvas at the given location, expanded or shunk to fit
+    * the width and height provided
+    * 
+    * @param file     the name of the file in the asset manager
+    * @param x        the left side of the image in the canvas
+    * @param y        the top of the image in the canvas
+    * @param width    the width to draw the image on the canvas
+    * @param height   the height to draw the image on the canvas
+    * @param rotation the amount to rotate the image
+    * @throws FileNotFoundException if the file can't be found in the project.
+    */
+   public static void drawImage(String filename, double x, double y, double width, double height, double rotation)
```

Comment:
As a general comment, all of these draw commands seem to be pretty complicated and take a lot of parameters that are the same type.  (This makes is easy to accidentally pass the wrong value for a parameter.)  Not sure how best to mitigate this, maybe we can have "simple" versions of these methods or make some of the parameters more stateful (similar to how fillColor is stateful)?

Reply:
Lots of discussion went back and forth on the design of these methods :) 

These are based on the Game Lab & App Lab drawing commands, which are in turn based on processing.org/p5js.org drawing commands. Those tools are widely used in introductory courses in Java and Javascript respectively. In my experience and reports I've heard from teachers that use those tools, the long list of identical parameters can be confusing at first but students get used to them pretty quickly. Stateful systems like the HTML5 canvas which use `moveTo` and `lineTo` have some other comprehension issues associated with them, and we decided that a (somewhat) stateless model for drawing would be best for this audience.

Reply:
Thanks for the context, sounds good to me!